### PR TITLE
feat: added voice and banner instructions

### PR DESCRIPTION
**Pull request by Archdoog**
- Adds `VoiceInstruction`
- Adds `BannerInstruction`

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -270,6 +278,58 @@ components:
...
+    VoiceInstruction:
+      type: object
+      properties:
+        distanceAlongGeometry:
+          type: number
+          format: double
+        announcement:
+          type: string
+        ssmlAnnouncement:
+          type: string
```
Comment (ianthetechie) [inline openapi.yaml:293]: I think that the `distanceAlongGeometry` and `announcement` types are required here.

File: openapi.yaml [hunk 2]
```diff
@@ -270,6 +278,58 @@ components:
...
+              type: string
+            components:
+              type: array
+              items:
+                type: object
+                properties:
+                  text:
+                    type: string
+                  type:
+                    type: string
```
Comment (ianthetechie) [inline openapi.yaml]: I'd suggest extracting this out into a `BannerContent` model as it's exactly the same for primary and secondary. `text` can also be a required field.

File: openapi.yaml [hunk 3]
```diff
@@ -270,6 +278,58 @@ components:
...
```
Comment (ianthetechie) [inline openapi.yaml]: On the `BannerInstructions` model, `primary` can be required.

**Review by ianthetechie — COMMENTED** [review]
I think I'm getting down into the weeds now, but want to get in the habit of marking required fields as such. This will become important for dev ergonomics as we start using it.

Unfortunately I don't use OSRM directly and it's not very well-documented, but empirically you can refer to what's in Ferrostar, as it has test data from both Valhalla and the public OSRM demo instance.

For the mapbox extensions (like these), we rely on finding the right balance of Mapbox's publicly documented APIs and Valhalla. I think Mapbox always ships SSML for example, but Valhalla never does, so I leave that as optional.

File: openapi.yaml (hunk 3 above)
Comment (Archdoog) [reply openapi.yaml]: Required this and the distance.

**Pull request merged**